### DISCUSSION TITLE
rnodeconf on Windows or rnspure on Linux - trouble, pls help :(

### DISCUSSION THREAD

**Discussion by rn1m** [markqvist/Reticulum#436] [Help & Questions] [answered]
Hello OMs!
I am a radio amateur, I want to flash the LoRa module board exclusively as a KISS TNC.
I tried to install rns on Linux, as described in the guide, but I always get the error that cryptoraphy cannot be installed completely (but I don’t need it).
Okay, I installed the rnspure package (successfully) but rnodeconf does not start, there is no such file for the system.
I thought maybe there was an option to install all packages on Windows?
Or after installing rnspure, how can you launch rnodeconf?
I'm new to Linux systems and looking for help.
Thank you in advance!

**Comment by rn1m** [comment]
I'm trying to install rnodeconf to flash LoRa modules in KISS modem mode. As a terminal - NanoPI + DietPi OS
Since encryption is not needed, I proceed by installing the rnspure package.
Main error: Can't install pyserial
It is installed via pipx, but rnodeconf does not see it. Can't install via pip3(
What can you recommend?

`pipx install rnspure
installed package rnspure 0.7.0, installed using Python 3.11.2
These apps are now globally available
- rncp
- rnid
- rnir
- rnodeconf
- rnpath
- rnprobe
- rnsd
- rnstatus
- rnx
done! ✨ 🌟 ✨
root@DietPi:~# rnodeconf

RNode Config Utility needs pyserial to work.
You can install it with: pip3 install pyserial

root@DietPi:~# pipx install pyserial
installed package pyserial 3.5, installed using Python 3.11.2
These apps are now globally available
- pyserial-miniterm
- pyserial-ports
done! ✨ 🌟 ✨
root@DietPi:~# rnodeconf --help

RNode Config Utility needs pyserial to work.
You can install it with: pip3 install pyserial

root@DietPi:~# pip3 install pyserial
error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try apt install
python3-xyz, where xyz is the package you are trying to
install.

```
If you wish to install a non-Debian-packaged Python package,
create a virtual environment using python3 -m venv path/to/venv.
Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
sure you have python3-full installed.

If you wish to install a non-Debian packaged Python application,
it may be easiest to use pipx install xyz, which will manage a
virtual environment for you. Make sure you have pipx installed.

See /usr/share/doc/python3.11/README.venv for more information.

```

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification.
`

**Comment by rn1m** [comment]
I continue to try to install rnodeconf^

root@DietPi:~# pipx inject rnspure pyserial

root@DietPi:~# rnodeconf --help

RNode Config Utility needs the cryptography module to work.
You can install it with: pip3 install cryptography

Do you need cryptography to flash the Laura module as TNC?
Next I do this:

root@DietPi:~# apt install python3-cryptography
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following additional packages will be installed:
  python3-cffi-backend
Suggested packages:
  python-cryptography-doc python3-cryptography-vectors
The following NEW packages will be installed:
  python3-cffi-backend python3-cryptography
0 upgraded, 2 newly installed, 0 to remove and 6 not upgraded.
Need to get 675 kB of archives.
After this operation, 2785 kB of additional disk space will be used.
Do you want to continue? [Y/n] Y
Get:1 https://deb.debian.org/debian bookworm/main armhf python3-cffi-backend armhf 1.15.1-5+b1 [75.1 kB]
Get:2 https://deb.debian.org/debian bookworm/main armhf python3-cryptography armhf 38.0.4-3 [600 kB]
Fetched 675 kB in 0s (1478 kB/s)
debconf: delaying package configuration, since apt-utils is not installed
Selecting previously unselected package python3-cffi-backend:armhf.
(Reading database ... 32704 files and directories currently installed.)
Preparing to unpack .../python3-cffi-backend_1.15.1-5+b1_armhf.deb ...
Unpacking python3-cffi-backend:armhf (1.15.1-5+b1) ...
Selecting previously unselected package python3-cryptography.
Preparing to unpack .../python3-cryptography_38.0.4-3_armhf.deb ...
Unpacking python3-cryptography (38.0.4-3) ...
Setting up python3-cffi-backend:armhf (1.15.1-5+b1) ...
Setting up python3-cryptography (38.0.4-3) ...


Again:

root@DietPi:~# rnodeconf --help

RNode Config Utility needs the cryptography module to work.
You can install it with: pip3 install cryptography

What am I doing wrong?

**Comment by rn1m** [comment, marked as answer]
I'm new to Linux, but thanks to the community I succeeded, below is the approximate order of command options, don't judge strictly. We managed to launch and flash the lore module as KISS TNC.

sudo apt install python3-pip
sudo apt install python3-full
sudo apt install pipx
sudo apt-get install build-essential libssl-dev libffi-dev \
python3-dev cargo pkg-config
pipx ensurepath
pipx install rns 

pipx install pyserial
;or pip3 install --break-system-packages --user pyserial
pipx inject rns pyserial

pipx install cryptography
; or sudo apt install python3-cryptography

pipx run --spec rns rnodeconf -h
pipx install rns --force

**Reply by markqvist** [reply]
Thanks a bunch for posting what worked out for you!
